src/Microsoft.SqlTools.ServiceLayer/QueryExecution/DataStorage/SaveAsExcelFileStreamWriterHelper.cs

```diff
@@ -0,0 +1,583 @@
+﻿/**
+ * A simple library to write xlsx file directly. It tries to be minimal, 
+ * both in implementation and runtime allocation.
+ * 
+ * A xlsx file is a zip with specific file structure.
+ * http://www.ecma-international.org/publications/standards/Ecma-376.htm
+ * 
+ * The page number is in the 
+ * ECMA-376, Fifth Edition, Part 1 - Fundamentals And Markup Language Reference 
+ * 
+ * Page 75
+ * /
+ * |- [Content_Types].xml
+ * |- _rels
+ *    |- .rels
+ * |- xl
+ *    |- workbook.xml
+ *    |- styles.xml
+ *    |- _rels
+ *       |- workbook.xml.rels
+ *    |- worksheets
+ *       |- sheet1.xml
+ * 
+ */
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Xml;
+
+namespace Microsoft.SqlTools.ServiceLayer.QueryExecution.DataStorage
+{
+    public sealed class SaveAsExcelFileStreamWriterHelper : IDisposable
```

Comment:
Please provide class-level block comments for public classes. I think in this case we can move the big block comment at the top of the file down here.